src/receive_imf.rs

```diff
@@ -966,52 +966,55 @@ async fn add_parts(
                         );
                     }
                 }
+            }
+        }
 
-                // Check if the message was sent with verified encryption and set the protection of
-                // the 1:1 chat accordingly.
-                let chat = match is_partial_download.is_none()
-                    && mime_parser.get_header(HeaderDef::SecureJoin).is_none()
-                    && !is_mdn
-                {
-                    true => Some(Chat::load_from_db(context, chat_id).await?)
-                        .filter(|chat| chat.typ == Chattype::Single),
-                    false => None,
+        if let Some(chat_id) = chat_id {
+            let contact = Contact::get_by_id(context, from_id).await?;
```

Comment:
Performance: `get_by_id()` loads the contact from the db, we should only do this if necessary.

What does this change do, anyway?

Reply:
We now unconditionally check if we want to upgrade the protection status instead of only doing it when we created a chat.

Reply:
Do we need this? If so, should probably go into an extra PR since it's unrelated?

Reply:
its needed for some test to succeed

Reply:
This is needed because now a 1:1 chat may have been looked up not in the block above, but also in the code block before, where `lookup_chat_or_create_adhoc_group()` is called. But still, the extra call to `Contact::get_by_id()` isn't needed in case if we're here after executing the code block above as it has already loaded the contact. So, this should be refactored somehow.

Reply:
agreed